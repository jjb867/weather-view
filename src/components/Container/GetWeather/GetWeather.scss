input.input:focus {
  box-shadow: none !important;
}

.is-small {
  margin-bottom: 1rem !important;
}

.card {
  border-radius: 4px;
}

.box {
  padding: 0.75rem;
}

.switch.on {
  background: green;
}

.switch {
  border: 1px solid #eee;
  width: 41px;
  height: 17px;
  border-radius: 13px;
  cursor: pointer;
  display: inline-block;
}

.switch-toggle {
  border: 1px solid #cec5c5;
  width: 19px;
  height: 19px;
  left: -3px;
  bottom: 2px;
  border-radius: 9px;
  background: #fff;
  position: relative;
  transition: left 100ms ease-in-out;
}

.switch.on .switch-toggle {
  left: 1.5rem;
}

.button.is-link.is-medium {
  //margin-left: 1rem;
}

.field.has-addons .control:not(:last-child) {
  margin-right: 0.5rem;
}

.field.has-addons .control:first-child .input {
  border-radius: 4px;
}

.mobile-switch {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.desktop-switch {
  margin-top: 0.67rem;
}

@media screen and (max-width: 768px) {
  .box:not(:last-child) {
    width: auto;
  }
  .field.has-addons .control:last-child .button,
  .field.has-addons .control:first-child .input {
    border-radius: 4px;
  }
  .section {
    padding-top: 1rem;
  }
  .is-size-6-mobile {
    font-size: 1rem !important;
    margin-top: 0.5rem;
  }
}



@media screen and (max-width: 500px) {
  h1.is-size-3 {
    font-size: 1.5rem !important;
  }
  p.attribution {
    font-size: 0.75rem !important;
  }
  //button.button.is-link,
  //input.input.is-medium {
  //  font-size: 1.125rem;
  //}
  p.has-text-info {
    font-size: 0.875rem;
  }
}
