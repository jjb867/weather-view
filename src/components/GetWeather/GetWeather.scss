input.input:focus {
    box-shadow: none !important;
}

button.is-small.is-link {
  margin-bottom: 0.5rem !important;
}

@media screen and (max-width: 500px) {
  h1.is-size-3 {
    font-size: 1.5rem !important;
  }
  p.attribution {
      font-size: 0.65rem !important;
  }
  button.button.is-link,
  input.input.is-medium {
    font-size: 1.125rem;
  }
}
