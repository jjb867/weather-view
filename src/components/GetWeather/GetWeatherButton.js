import React from 'react';
import { Button } from '../../utils/poseAnimations';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const GetWeatherButton = props => {
    return (
        <Button className="button is-link is-medium" onClick={props.onClick}>
            <FontAwesomeIcon icon="cloud-sun-rain"/>&nbsp;
            Get Weather
        </Button>
    );
};

export default GetWeatherButton;
