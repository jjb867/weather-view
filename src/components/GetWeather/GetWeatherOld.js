{/*<section className="section" onKeyPress={this.onKeyPress}>*/}
{/*<div className="container">*/}
{/*<section className="columns">*/}
{/*<div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">*/}
{/*<h1 className="is-size-3">{this.state.searchTitle}</h1>*/}
{/*<p className="attribution">*/}
{/*Powered by{' '}*/}
{/*<a href={'https://openweathermap.org'}>https://openweathermap.org</a>*/}
{/*</p>*/}
{/*</div>*/}
{/*</section>*/}
{/*<section className="columns">*/}
{/*<div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">*/}
{/*<SearchBar*/}
{/*name="searchTerm"*/}
{/*value={this.state.searchTerm}*/}
{/*onChange={this.onSearchTermChange}*/}
{/*/>*/}
{/*</div>*/}
{/*</section>*/}
{/*<section className="columns">*/}
{/*<div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">*/}
{/*<GetWeatherButton onClick={this.onClick} />*/}
{/*</div>*/}
{/*</section>*/}
{/*<section className="columns">*/}
{/*<div className="column is-10-mobile is-offset-1-mobile column is-8 is-offset-2">*/}
{/*<button*/}
{/*className="button is-small is-link"*/}
{/*onClick={this.onBackToSearchClick}*/}
{/*>*/}
{/*Back to Search*/}
{/*</button>*/}
{/*<WeatherCardAnimation*/}
{/*key="animation"*/}
{/*className="animation"*/}
{/*pose={_isResultVisible ? 'visible' : 'hidden'}*/}
{/*>*/}
{/*<WeatherCard weather={currentWeather} />*/}
{/*</WeatherCardAnimation>*/}
{/*</div>*/}
{/*// </section>*/}

{/*</div>*/}
{/*// </section>*/}
{/*);*/}