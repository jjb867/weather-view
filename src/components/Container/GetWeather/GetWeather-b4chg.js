import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './GetWeather.scss';
import { connect } from 'react-redux';
import Switch from 'react-toggle-switch';
import {
    getCurrentWeatherByZip,
    getCurrentWeatherByCity,
    clearWeather,
    getForecastByCity,
    getForecastByZip,
    setError,
    clearError,
} from '../../../store/actions/weatherActions';
import { zipRegex, cityRegex } from '../../../utils/regex';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { GetWeatherAnimation, GetForecastAnimation, Button, charPoses } from '../../../utils/poseAnimations';
import SplitText from 'react-pose-text';
import SearchBar from '../../SearchBar/SearchBar';
import GetWeatherButton from '../../GetWeatherButton/GetWeatherButton';
import WeatherCard from '../../WeatherCard/WeatherCard';
import Forecast from '../../Forecast/Forecast';

class GetWeather extends Component {
    constructor() {
        super();
        this.state = {
            searchTerm: '',
            invalidSearch: '',
            invalidMinLength: '',
            invalidMaxLength: '',
            selectedSearchParameter: '',
            poweredByUrl: 'https://openweathermap.org',
            invalidSearchText: 'Enter a valid City or Zip',
            isSearchFadedIn: false,
            isSearchFocused: false,
            isForecastToggled: false,
        };
    }

    componentDidMount() {
        this.setState({ isSearchFadedIn: true });
    }

    onSearchTermChange = e => {
        this.setState({
            searchTerm: e.target.value,
        });
    };

    onBackToSearchClick = () => {
        this.props.clearWeather();
    };

    checkForValidSearchCity = searchTerm => {
        return cityRegex.test(searchTerm);
    };

    checkForValidSearchZip = searchTerm => {
        return zipRegex.test(searchTerm);
    };

    onKeyPress = e => {
        if (e.key === 'Enter') {
            this.getWeather();
        }
    };

    getWeather = () => {
        const { searchTerm, isForecastToggled, invalidSearchText } = this.state;
        const {
            getCurrentWeatherByCity,
            getForecastByCity,
            getCurrentWeatherByZip,
            getForecastByZip,
            clearError,
            setError,
        } = this.props;

        if (this.checkForValidSearchCity(searchTerm)) {
            if(!isForecastToggled) {
                getCurrentWeatherByCity(searchTerm)
            } else {
                getForecastByCity(searchTerm);
            }
            clearError();
        } else if (this.checkForValidSearchZip(searchTerm)) {
            !isForecastToggled ? getCurrentWeatherByZip(searchTerm) : getForecastByZip(searchTerm);
            clearError();
        } else {
            setError(invalidSearchText);
        }
    };

    onClick = () => {
        const { length } = this.state.searchTerm;
        const { setError } = this.props;
        ((length > 2) && (length < 50)) ? this.getWeather() : setError(this.state.invalidSearchText);
    };

    onSearchBarFocus = () => {
        this.setState({ isSearchFocused: true });
    };

    onGetWeatherClick = e => {
        if (!e.target.classList.contains('search-bar')) {
            this.setState({ isSearchFocused: false });
        }
    };

    isWeatherResultEmpty = obj => {
        for (let key in obj) {
            if (obj.hasOwnProperty(key)) return false;
        }
        return true;
    };

    onToggleSwitchClick = () => {
        this.props.clearError();
        this.setState(prevState => {
            return {
                isForecastToggled: !prevState.isForecastToggled,
            };
        });
    };

    render() {
        const { currentWeather, forecast } = this.props.weather;
        const { message } = this.props.error;
        const { isSearchFadedIn, isForecastToggled, searchTerm, poweredByUrl } = this.state;
        const currentWeatherEmpty = this.isWeatherResultEmpty(currentWeather);
        const forecastEmpty = this.isWeatherResultEmpty(forecast);
        console.log('render called get Weather');
        return (
            <React.Fragment>
                <GetWeatherAnimation
                    key="search"
                    className="search"
                    initialPose={isSearchFadedIn ? 'visible' : 'hidden'}
                    pose={currentWeatherEmpty && forecastEmpty ? 'visible' : 'hidden'}
                >
                    {currentWeatherEmpty && forecastEmpty && (
                        <section className="section" onKeyPress={this.onKeyPress} onClick={this.onGetWeatherClick}>
                            <div className="container">
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <h1 className="is-size-2">WeatherView</h1>
                                        <h3 className="is-size-5">
                                            <SplitText initialPose="exit" pose="enter" charPoses={charPoses}>
                                                {!isForecastToggled ? 'Current Weather' : '24 Hour Forecast'}
                                            </SplitText>
                                        </h3>
                                        <p className="attribution">
                                            Powered by <a href={poweredByUrl}>{poweredByUrl}</a>
                                        </p>
                                    </div>
                                </div>
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <SearchBar
                                            name="searchTerm"
                                            value={searchTerm}
                                            onChange={this.onSearchTermChange}
                                            onFocus={this.onSearchBarFocus}
                                            errorMessage={message}
                                        />
                                    </div>
                                </div>
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <div className="columns is-gapless">
                                            <div className="column is-3">
                                                <label>
                                                    <SplitText initialPose="exit" pose="enter" charPoses={charPoses}>
                                                        {!isForecastToggled ? 'Current Weather' : '24 Hour Forecast'}
                                                    </SplitText>
                                                </label>
                                            </div>
                                            <div className="column is-1">
                                                <Switch
                                                    className="has-background-link"
                                                    onClick={this.onToggleSwitchClick}
                                                    on={isForecastToggled}
                                                />
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">

                                    </div>
                                </div>
                            </div>
                        </section>
                    )}
                </GetWeatherAnimation>

                <GetWeatherAnimation
                    key="weather"
                    className="weather"
                    pose={!currentWeatherEmpty && forecastEmpty ? 'visible' : 'hidden'}
                >
                    {!currentWeatherEmpty && forecastEmpty && (
                        <section className="section">
                            <div className="container">
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <Button className="button is-small is-link" onClick={this.onBackToSearchClick}>
                                            <FontAwesomeIcon icon="arrow-left" />
                                            &nbsp;Back to Search
                                        </Button>
                                        <WeatherCard weather={currentWeather} />
                                    </div>
                                </div>
                            </div>
                        </section>
                    )}
                </GetWeatherAnimation>

                <GetForecastAnimation
                    key="forecast"
                    className="forecast"
                    pose={currentWeatherEmpty && !forecastEmpty ? 'visible' : 'hidden'}
                >
                    {currentWeatherEmpty && !forecastEmpty && (
                        <div className="column">
                            <Button className="button is-small is-link" onClick={this.onBackToSearchClick}>
                                <FontAwesomeIcon icon="arrow-left" />
                                &nbsp; Back to Search
                            </Button>
                            <Forecast forecast={forecast} />
                        </div>
                    )}
                </GetForecastAnimation>
            </React.Fragment>
        );
    }
}

const mapStateToProps = state => ({
    searchTerm: state.searchTerm,
    weather: state.weather,
    error: state.error,
});

GetWeather.propTypes = {
    getCurrentWeatherByZip: PropTypes.func.isRequired,
    getCurrentWeatherByCity: PropTypes.func.isRequired,
    clearWeather: PropTypes.func.isRequired,
    getForecastByZip: PropTypes.func.isRequired,
    getForecastByCity: PropTypes.func.isRequired,
    setError: PropTypes.func.isRequired,
    clearError: PropTypes.func.isRequired,
};

export default connect(
    mapStateToProps,
    {
        getCurrentWeatherByZip,
        getCurrentWeatherByCity,
        clearWeather,
        getForecastByCity,
        getForecastByZip,
        setError,
        clearError,
    },
)(GetWeather);
