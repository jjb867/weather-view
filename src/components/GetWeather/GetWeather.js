import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Dashboard.scss';
import posed from 'react-pose';
import { connect } from 'react-redux';
import { getCurrentWeatherByZip, getCurrentWeatherByCity } from '../../store/actions/weatherActions';
import { zipRegex, cityRegex } from '../../configuration/regex';
import SearchBar from './SearchBar/SearchBar';
import GetWeatherButton from './GetWeatherButton';
import WeatherCard from './WeatherCard/WeatherCard';

// Pose Animations
const WeatherCardAnimation = posed.div({
    hidden: { y: -10, x: 10, opacity: 0, transition: { duration: 200 } },
    visible: { y: 0, x: 0, opacity: 1, transition: { duration: 200 } },
});

class GetWeather extends Component {
    constructor() {
        super();
        this.state = {
            searchTerm: '',
            selectedSearchParameter: '',
            searchTitle: 'Current Weather',
            isResultVisible: false,
        };
    }

    onSearchTermChange = e => {
        this.setState({
            searchTerm: e.target.value,
        });
    };

    onBackToSearchClick = () => {
        console.log('on back to search click');
        this.setState({ isResultVisible: false });
    };

    checkForValidSearchCity = searchTerm => {
        return cityRegex.test(searchTerm);
    };

    checkForValidSearchZip = searchTerm => {
        return zipRegex.test(searchTerm);
    };

    onKeyPress = e => {
        if (e.key === 'Enter') {
            this.getWeather();
        }
    };

    getWeather = () => {
        if (this.checkForValidSearchCity(this.state.searchTerm)) {
            this.props.getCurrentWeatherByCity(this.state.searchTerm);
        } else if (this.checkForValidSearchZip(this.state.searchTerm)) {
            this.props.getCurrentWeatherByZip(this.state.searchTerm);
        } else {
            alert('handle ERROR: invalid search term');
        }
    };

    onClick = () => {
        if (this.state.searchTerm.length > 2) {
            this.getWeather();
        } else {
            alert('Must be at least 3 characters');
        }
    };

    isWeatherResultEmpty = obj => {
        for (let key in obj) {
            if (obj.hasOwnProperty(key)) return false;
        }
        return true;
    };

    render() {
        const { currentWeather } = this.props.weather;

        let WeatherContent;

        const weatherContentRenderer = currentWeather => {
            if (this.isWeatherResultEmpty(currentWeather)) {
                return (
                    <section className="section" onKeyPress={this.onKeyPress}>
                        <div className="container">
                            <section className="columns">
                                <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                    <h1 className="is-size-3">{this.state.searchTitle}</h1>
                                    <p className="attribution">
                                        Powered by <a href={'https://openweathermap.org'}>https://openweathermap.org</a>
                                    </p>
                                </div>
                            </section>
                            <section className="columns">
                                <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                    <GetWeatherButton onClick={this.onClick} />
                                </div>
                            </section>
                        </div>
                    </section>
                );
            } else {
                return (
                    <section className="columns">
                        <div className="column is-10-mobile is-offset-1-mobile column is-8 is-offset-2">
                            <button className="button is-small is-link" onClick={this.onBackToSearchClick}>
                                Back to Search
                            </button>
                            {/*<WeatherCardAnimation*/}
                            {/*key="animation"*/}
                            {/*className="animation"*/}
                            {/*pose={_isResultVisible ? 'visible' : 'hidden'}*/}
                            {/*>*/}
                            <WeatherCard weather={currentWeather} />
                            {/*</WeatherCardAnimation>*/}
                        </div>
                    </section>
                );
            }
        };

        WeatherContent = weatherContentRenderer(currentWeather);

        return WeatherContent;
    }
}

const mapStateToProps = state => ({
    searchTerm: state.searchTerm,
    weather: state.weather,
});

GetWeather.propTypes = {
    getCurrentWeatherByZip: PropTypes.func.isRequired,
    getCurrentWeatherByCity: PropTypes.func.isRequired,
};

export default connect(
    mapStateToProps,
    { getCurrentWeatherByZip, getCurrentWeatherByCity },
)(GetWeather);
