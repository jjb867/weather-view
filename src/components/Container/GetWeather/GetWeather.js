import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './GetWeather.scss';
import { connect } from 'react-redux';
import Switch from 'react-toggle-switch';
import {
    getCurrentWeatherByZip,
    getCurrentWeatherByCity,
    clearWeather,
    getForecastByCity,
    getForecastByZip,
    setError,
    clearError,
} from '../../../store/actions/weatherActions';
import { zipRegex, cityRegex } from '../../../utils/regex';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { GetWeatherAnimation, GetForecastAnimation, Button, charPoses } from '../../../utils/poseAnimations';
import SplitText from 'react-pose-text';
import SearchBar from '../../SearchBar/SearchBar';
import GetWeatherButton from '../../GetWeatherButton/GetWeatherButton';
import WeatherCard from '../../WeatherCard/WeatherCard';
import Forecast from '../../Forecast/Forecast';

class GetWeather extends Component {
    constructor() {
        super();
        this.state = {
            searchTerm: '',
            invalidSearch: '',
            invalidMinLength: '',
            invalidMaxLength: '',
            selectedSearchParameter: '',
            poweredByUrl: 'https://openweathermap.org',
            isSearchFadedIn: false,
            isSearchFocused: false,
            isForecastToggled: false,
        };
    }

    componentDidMount() {
        this.setState({ isSearchFadedIn: true });
    }

    onSearchTermChange = e => {
        this.setState({
            searchTerm: e.target.value,
        });
    };

    onBackToSearchClick = () => {
        this.props.clearWeather();
    };

    checkForValidSearchCity = searchTerm => {
        return cityRegex.test(searchTerm);
    };

    checkForValidSearchZip = searchTerm => {
        return zipRegex.test(searchTerm);
    };

    onKeyPress = e => {
        if (e.key === 'Enter') {
            this.getWeather();
        }
    };

    getWeather = () => {
        const { searchTerm, isForecastToggled } = this.state;
        const {
            getCurrentWeatherByCity,
            getForecastByCity,
            getCurrentWeatherByZip,
            getForecastByZip,
            clearError,
            setError,
        } = this.props;

        if (this.checkForValidSearchCity(searchTerm)) {
            !isForecastToggled ? getCurrentWeatherByCity(searchTerm) : getForecastByCity(searchTerm);
            clearError();
        } else if (this.checkForValidSearchZip(searchTerm)) {
            !isForecastToggled ? getCurrentWeatherByZip(searchTerm) : getForecastByZip(searchTerm);
            clearError();
        } else {
            setError('Invalid search term');
        }
    };

    onClick = () => {
        const { length } = this.state.searchTerm;
        const { setError } = this.props;
        if (length > 2) {
            this.getWeather();
        } else if (length > 50) {
            setError('Search term must not be more than 50 characters');
        } else {
            setError('Search term must be at least 3 characters');
        }
    };

    onSearchBarFocus = () => {
        this.setState({ isSearchFocused: true });
    };

    onGetWeatherClick = e => {
        if (!e.target.classList.contains('search-bar')) {
            this.setState({ isSearchFocused: false });
        }
    };

    isWeatherResultEmpty = obj => {
        for (let key in obj) {
            if (obj.hasOwnProperty(key)) return false;
        }
        return true;
    };

    onToggleSwitchClick = () => {
        this.props.clearError();
        this.setState(prevState => {
            return {
                isForecastToggled: !prevState.isForecastToggled,
                searchTerm: ''
            };
        });
    };

    render() {
        const { currentWeather, forecast } = this.props.weather;
        const { message } = this.props.error;
        const { isSearchFadedIn, isForecastToggled, searchTerm, poweredByUrl } = this.state;
        const currentWeatherEmpty = this.isWeatherResultEmpty(currentWeather);
        const forecastEmpty = this.isWeatherResultEmpty(forecast);

        return (
            <React.Fragment>
                <GetWeatherAnimation
                    key="search"
                    className="search"
                    initialPose={isSearchFadedIn ? 'visible' : 'hidden'}
                    pose={(currentWeatherEmpty && forecastEmpty) ? 'visible' : 'hidden'}
                >
                    {(currentWeatherEmpty && forecastEmpty) && (
                        <section className="section" onKeyPress={this.onKeyPress} onClick={this.onGetWeatherClick}>
                            <div className="container">
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <h3 className="is-size-3">
                                            <SplitText initialPose="exit" pose="enter" charPoses={charPoses}>
                                                {!isForecastToggled ? 'Current Weather' : '24 Hour Forecast'}
                                            </SplitText>
                                        </h3>
                                        <p className="attribution">
                                            Powered by <a href={poweredByUrl}>{poweredByUrl}</a>
                                        </p>
                                    </div>
                                </div>
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <SearchBar
                                            name="searchTerm"
                                            value={searchTerm}
                                            onChange={this.onSearchTermChange}
                                            onFocus={this.onSearchBarFocus}
                                            errorMessage={message}
                                        />
                                    </div>
                                </div>
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <div className="columns is-gapless">
                                            <div className="column is-3">
                                                <label>
                                                    <SplitText initialPose="exit" pose="enter" charPoses={charPoses}>
                                                        {!isForecastToggled ? 'Current Weather' : '24 Hour Forecast'}
                                                    </SplitText>
                                                </label>
                                            </div>
                                            <div className="column is-1">
                                                <Switch
                                                    className="has-background-link"
                                                    onClick={this.onToggleSwitchClick}
                                                    on={isForecastToggled}
                                                />
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <GetWeatherButton onClick={this.onClick} />
                                    </div>
                                </div>
                            </div>
                        </section>
                    )}
                </GetWeatherAnimation>
                <GetForecastAnimation
                    key="weather"
                    className="weather"
                    pose={(!currentWeatherEmpty && forecastEmpty) ? 'visible' : 'hidden'}
                >
                    {(!currentWeatherEmpty && forecastEmpty) && (
                        <section className="section">
                            <div className="container">
                                <div className="columns">
                                    <div className="column is-10-mobile is-offset-1-mobile is-8 is-offset-2">
                                        <Button className="button is-small is-link" onClick={this.onBackToSearchClick}>
                                            <FontAwesomeIcon icon="arrow-left" />
                                            &nbsp; Back to Search
                                        </Button>
                                        <WeatherCard weather={currentWeather} />
                                    </div>
                                </div>
                            </div>
                        </section>
                    )}
                </GetForecastAnimation>
                <GetWeatherAnimation
                    key="forecast"
                    className="forecast"
                    pose={(currentWeatherEmpty && !forecastEmpty) ? 'visible' : 'hidden'}
                >
                    {(currentWeatherEmpty && !forecastEmpty) && (
                        <section className="section">
                            <div className="container">
                                <div className="columns">
                                    <div className="column is-12">
                                        <Button className="button is-small is-link" onClick={this.onBackToSearchClick}>
                                            <FontAwesomeIcon icon="arrow-left" />
                                            &nbsp; Back to Search
                                        </Button>
                                        <Forecast forecast={forecast} />
                                    </div>
                                </div>
                            </div>
                        </section>
                    )}
                </GetWeatherAnimation>
            </React.Fragment>
        );
    }
}

const mapStateToProps = state => ({
    searchTerm: state.searchTerm,
    weather: state.weather,
    error: state.error,
});

GetWeather.propTypes = {
    getCurrentWeatherByZip: PropTypes.func.isRequired,
    getCurrentWeatherByCity: PropTypes.func.isRequired,
    clearWeather: PropTypes.func.isRequired,
    getForecastByZip: PropTypes.func.isRequired,
    getForecastByCity: PropTypes.func.isRequired,
    setError: PropTypes.func.isRequired,
    clearError: PropTypes.func.isRequired,
};

export default connect(
    mapStateToProps,
    {
        getCurrentWeatherByZip,
        getCurrentWeatherByCity,
        clearWeather,
        getForecastByCity,
        getForecastByZip,
        setError,
        clearError,
    }
)(GetWeather);
