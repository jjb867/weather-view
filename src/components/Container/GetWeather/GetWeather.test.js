import React from 'react';
import { shallow, mount } from 'enzyme';
import GetWeather from './GetWeather';

describe('GetWeather', () => {
    it('should render without crashing', () => {
        const component = shallow(<GetWeather />);
        expect(component).toMatchSnapshot();
    });
});