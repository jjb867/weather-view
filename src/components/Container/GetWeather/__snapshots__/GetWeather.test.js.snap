// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetWeather should render without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
